demos/python_demos/segmentation_demo/segmentation_demo.py

```diff
@@ -15,199 +15,229 @@
  limitations under the License.
 """
 
+import logging
+import random
 import sys
-import os
 from argparse import ArgumentParser, SUPPRESS
+from pathlib import Path
+from time import perf_counter
+import os
+
 import cv2
 import numpy as np
-import logging as log
 from openvino.inference_engine import IECore
 
-classes_color_map = [
-    (150, 150, 150),
-    (58, 55, 169),
-    (211, 51, 17),
-    (157, 80, 44),
-    (23, 95, 189),
-    (210, 133, 34),
-    (76, 226, 202),
-    (101, 138, 127),
-    (223, 91, 182),
-    (80, 128, 113),
-    (235, 155, 55),
-    (44, 151, 243),
-    (159, 80, 170),
-    (239, 208, 44),
-    (128, 50, 51),
-    (82, 141, 193),
-    (9, 107, 10),
-    (223, 90, 142),
-    (50, 248, 83),
-    (178, 101, 130),
-    (71, 30, 204)
+sys.path.append(str(Path(__file__).resolve().parents[1] / 'common'))
+
+from models import SegmentationModel
+import monitors
+from pipelines import AsyncPipeline
+from images_capture import open_images_capture
+from performance_metrics import PerformanceMetrics
+
+logging.basicConfig(format='[ %(levelname)s ] %(message)s', level=logging.INFO, stream=sys.stdout)
+log = logging.getLogger()
+
+
+default_color_map = [
+    (0, 0, 0),
+    (128, 0, 0),
+    (0, 128, 0),
+    (128, 128, 0),
+    (0, 0, 128),
+    (128, 0, 128),
+    (0, 128, 128),
+    (128, 128, 128),
+    (64, 0, 0),
+    (192, 0, 0),
+    (64, 128, 0),
+    (192, 128, 0),
+    (64, 0, 128),
+    (192, 0, 128),
+    (64, 128, 128),
+    (192, 128, 128),
+    (0, 64, 0),
+    (128, 64, 0),
+    (0, 192, 0),
+    (128, 192, 0),
+    (0, 64, 128)
 ]
 
-np.random.seed(10)
+
+def create_color_map(input):
+    global default_color_map
+    global color_map
+
+    if input and os.path.isfile(input):
```

Comment:
```suggestion
    if input:
```
The demo should report an error (which `open()` does) if `input` isn't empty but not a file.